x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationReloaderTests.java

```diff
@@ -503,6 +506,27 @@ public void testPEMTrustReloadException() throws Exception {
         assertThat(sslService.sslContextHolder(config).sslContext(), sameInstance(context));
     }
 
+    /**
+     * Tests that the reloader doesn't throw an exception if a file is configured to be outside of the config/ directory.
+     * These errors are handled correctly by the
+     */
+    public void testReadingOutsideConfigDirectoryDoesntFail() throws Exception {
+        Path tempDir = createTempDir();
+        Path clientCertPath = tempDir.resolve("testclient.crt");
+        Settings settings = baseKeystoreSettings(tempDir, null)
+            .putList("xpack.security.transport.ssl.certificate_authorities", clientCertPath.toString())
+            .put("path.home", createTempDir())
+            .build();
+        Environment env = TestEnvironment.newEnvironment(settings);
+        final AtomicReference<Exception> exceptionRef = new AtomicReference<>();
+
+        final ResourceWatcherService mockResourceWatcher = Mockito.mock(ResourceWatcherService.class);
+        Mockito.when(mockResourceWatcher.add(Mockito.any(), Mockito.any())).thenThrow(new AccessControlException("access denied in test"));
+        final Collection<SSLConfiguration> configurations = SSLService.getSSLConfigurations(settings).values();
+        final SSLConfigurationReloader reloader = new SSLConfigurationReloader(env, null, mockResourceWatcher, configurations);
+        assertNotNull(reloader);
```

Comment:
Nit: isn't this redundant since the above constructor would have failed already if the object does not get created?